### ISSUE TITLE
Mouse pointer is off.

### ISSUE THREAD

**Issue by lucmp** [osxmidi/LinVst#230]
I have been rebuilding a new Linux installation. Began to use lin-vst on it today. I was using wine-staging 5.12, now I'm using wine-staging 9.22.

The mouse pointer is off. First I tried to add a Windows plugin to qtractor and couldn't click and move/tweak anything on the plugin's GUI. Then I tried Waveform and noticed the same problem. It took me a while to realize that the mouse pointer still works, but it's sending the point-and-click commands to a spot about two inches higher and two inches to the left.

I tried loading the .dll directly on a Windows program and the mouse action was correct. It's also correct if I load the plugin with testvst32.exe.

I am using LinVst-4.9. The LinVst-4.9-wine-9.5-9.6-fix version doesn't work at all, the DAW says the plugin can't be found.

Thank you for your attention.

**Comment by lucmp** [comment]
Maybe I should point out that at first I installed Wine stable 9.0 and there was no problem with lin-vst, but multiple plugins and applications didn't work until I replaced it with Wine staging 9.22. Now they work, but lin-vst has the mouse problem.

**Comment by osxmidi** [comment]
> I have been rebuilding a new Linux installation. Began to use lin-vst on it today. I was using wine-staging 5.12, now I'm using wine-staging 9.22.
> 
> The mouse pointer is off. First I tried to add a Windows plugin to qtractor and couldn't click and move/tweak anything on the plugin's GUI. Then I tried Waveform and noticed the same problem. It took me a while to realize that the mouse pointer still works, but it's sending the point-and-click commands to a spot about two inches higher and two inches to the left.
> 
> I tried loading the .dll directly on a Windows program and the mouse action was correct. It's also correct if I load the plugin with testvst32.exe.
> 
> I am using LinVst-4.9. The LinVst-4.9-wine-9.5-9.6-fix version doesn't work at all, the DAW says the plugin can't be found.
> 
> Thank you for your attention.

I've put up a possible test fix for wine 9.22 and wine 10 in the Releases section.

**Comment by Ramen2X** [comment]
> I've put up a possible test fix for wine 9.22 and wine 10 in the Releases section.

Unfortunately, it seems like I can still reproduce this issue with wine 9.22 using the `
LinVst-4.9-wine-9.22-wine-10-fix2.zip` in the Releases tab.

**Comment by osxmidi** [comment]
> > I've put up a possible test fix for wine 9.22 and wine 10 in the Releases section.
> 
> Unfortunately, it seems like I can still reproduce this issue with wine 9.22 using the `LinVst-4.9-wine-9.22-wine-10-fix2.zip` in the Releases tab.

Try installing it again, I've updated the wine 9.22 release in the Releases section.

Make sure that lin-vst-server.exe and lin-vst-server.exe.so and lin-vst-server32.exe and lin-vst-server32.exe are actually copied to the /usr/bin directory (using sudo).

Choose Update All Plugins (Upgrade All Plugins) when using linvstconvert.

**Comment by Ramen2X** [comment]
> Try installing it again, I've updated the wine 9.22 release in the Releases section.

Unfortunately, I'm still seeing the same issue with `fix3`. I fully reinstalled LinVST and all of my plugins as an extra measure to ensure nothing from the previous version was lingering, but it doesn't seem to make a difference.

**Comment by osxmidi** [comment]
> > Try installing it again, I've updated the wine 9.22 release in the Releases section.
> 
> Unfortunately, I'm still seeing the same issue with `fix3`. I fully reinstalled LinVST and all of my plugins as an extra measure to ensure nothing from the previous version was lingering, but it doesn't seem to make a difference.

You probably need to manually delete the lin-vst-server.exe and lin-vst-server.exe.so and lin-vst-server32.exe and lin-vst-server32.exe.so files in /usr/bin and check that they have been deleted (ls /usr/bin/lin-v*).

Then install the new LinVst-wine-9.22-wine10  lin-vst-server.exe and lin-vst-server.exe.so and lin-vst-server32.exe and lin-vst-server32.exe.so files into /usr/bin.

So far I've tested the LinVst-wine-9.22-wine10 version using EndeavourOS (Arch), MX Linux, and Ubuntu and wine 9, wine 9.22 and wine 10 rc2 and rc3 and it seems to work ok.

**Comment by Ramen2X** [comment]
> So far I've tested the LinVst-wine-9.22-wine10 version using EndeavourOS (Arch), MX Linux, and Ubuntu and wine 9, wine 9.22 and wine 10 rc2 and rc3 and it seems to work ok.

You're right, `fix3` does work. I can confirm my plugins work as expected when I load them through `testvst.exe`. The actual issue I'm facing seems to be a problem with my DAW itself, where the mouse cursor disappears and I can no longer interact with the plugin window. Sorry about that!

**Comment by dkdjam** [comment]
Not related to linvst / qtractor, but just to add to the original issue, I am experiencing the exact same problem with yabridge and ardour 8.11. I recently updated one of my VST plugins (Addictive Drums) and while I can use it completely fine as a standalone program or VST via `fsthost-1.6.3`, I can't get it to work properly any longer in ardour, observing the same weird out-of-focus mouse and unresponsive GUI buttons. I tried different wine versions (10.x) and 9.22 staging is the one where I landed (tried 10.2, nothing would work at all overall, and 10.1 was no different from 9.22). A real shame as I have many ardour sessions using this VST. Not sure what changed but I might see if older wine staging can make a diff. I have another PC in my rehearsal room with a working combo I ha not touched for a while so I might as well revert to that.

**Comment by PennRobotics** [comment]
A fix is coming shortly for yabridge, robbert-vdh/yabridge#405

**Comment by davkdavk** [comment]
I have a similar issue.

I am on Debian 13 with Wine 10 and have installed the latest linvst2/3. All of my plugins are ok except for Black Salt Audio - Silencer. When docked in the FX window, none of the controls can be clicked on.  

I have to float the FX window in Reaper and move it to the top left of screen to be able to move the controls, and i have to point just above the control for it to work and the alignment issues are the same as the original post. 

All other effects this far have been fine.
